## Exception

### 异常类型

自己看

IndexError

AssertionError

KeyError



### try-except语句

~~~python
try:
    #检测范围
except Exception[as reason]:
    #异常后的处理代码
~~~

例如,因为这个文件没有，所以肯定会抛出异常，可以用语句来处理

~~~python
try:
	f = open('没有这个文件.txt')
	print(f.read())
	f.close()
except OSError:
    print('文件出错')
~~~

更加详细可以这个样

~~~python
try:
	f = open('没有这个文件.txt')
	print(f.read())
	f.close()
except OSError as reason:
    print('文件出错','错误原因：'+ str(reason))
~~~

#### 可以一次处理多个异常

但是try到异常以后，后续语句不会执行

~~~python
try:
    int('abc')
    sum = 1 + '1'
	f = open('没有这个文件.txt')
	print(f.read())
	f.close()
except OSError as reason:
    print('文件出错','错误原因：'+ str(reason))
except TypeError as reason:
    print('类型出错','错误原因：'+ str(reason))  
~~~

或者这样

~~~python
try:
    int('abc')
    sum = 1 + '1'
	f = open('没有这个文件.txt')
	print(f.read())
	f.close()
except (OSError, TypeError) :
    print('出错')
~~~



### try-finally

~~~python
try:
    #检测范围
except Excetpion [as reason] :
    #异常处理代码
finally:
    #不管是否出错都会运行的代码
~~~



